core/controllers/question_test.py

```diff
@@ -228,128 +97,96 @@ def test_delete(self):
+            '%s' % feconf.QUESTION_CREATION_URL, {'payload': (
+                json.dumps(payload))}, expect_errors=True,
             expected_status_int=404)
 
-        del self.payload['question']
+        del payload['question']
         self.post_json(
-            '%s' % feconf.QUESTION_POST_URL, {'payload': (
-                json.dumps(self.payload))}, expect_errors=True,
+            '%s' % feconf.QUESTION_CREATION_URL, {'payload': (
+                json.dumps(payload))}, expect_errors=True,
             expected_status_int=404)
 
+        self.logout()
+        response = self.testapp.post(
+            '%s' % feconf.QUESTION_CREATION_URL, payload,
+            expect_errors=True)
+        self.assertEqual(response.status_int, 302)
 
-class QuestionManagerHandlerTest(test_utils.GenericTestBase):
-    """Test the question handler."""
+    def test_batch_get(self):
+        """Tests get method of questions batch handler."""
+        question_id = question_services.add_question(
+            self.owner_id, self.question)
+        self.question = question_services.get_question_by_id(question_id)
+        question_services.add_question_id_to_skill(
+            self.question.question_id, self.question.collection_id,
+            self.skill_id, self.owner_id)
 
-    def setUp(self):
-        super(QuestionManagerHandlerTest, self).setUp()
+        collection_services.record_played_exploration_in_collection_context(
+            self.new_user_id, self.collection_id, self.exp_id)
 
-        self.collection_id = 'coll_0'
-        self.exp_id = 'exp_1'
-        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
-        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+        payload = {}
+        payload['collection_id'] = self.collection_id
+        payload['stringified_skill_ids'] = json.dumps(
+            [self.skill_id, 'test'])
 
-        # Create a new collection and exploration.
-        self.save_new_valid_collection(
-            self.collection_id, self.owner_id, exploration_id=self.exp_id)
+        self.login(self.NEW_USER_EMAIL)
+        response_json = self.get_json(
+            '%s/batch' % feconf.QUESTION_DATA_URL, payload,
+            expect_errors=False)
+        self.assertIn(self.question.to_dict(), response_json['questions_dict'])
+        self.assertEqual(len(response_json['questions_dict']), 1)
 
-        # Add a skill.
-        collection_services.update_collection(
-            self.owner_id, self.collection_id, [{
-                'cmd': collection_domain.CMD_ADD_COLLECTION_SKILL,
-                'name': 'test'
-            }], 'Add a new skill')
-        collection = collection_services.get_collection_by_id(
-            self.collection_id)
-        self.skill_id = collection.get_skill_id_from_skill_name('test')
-        collection_node = collection.get_node(self.exp_id)
-        collection_node.update_acquired_skill_ids([self.skill_id])
+        response = self.testapp.get(
```

Comment:
Here and elsewhere -- I thought we talked about using self.get_json() instead of self.testapp.get()?

Please check the rest of the PR and fix accordingly.

Reply:
I think I have mentioned the reason for using self.testapp.get in the previous comments.
So, whenever I try using get_json for the failing conditions, I get the following error:
`Traceback (most recent call last):
  File "/Users/giritheja/Desktop/opensource/oppia/core/controllers/question_test.py", line 164, in test_batch_get
    expect_errors=True)
  File "/Users/giritheja/Desktop/opensource/oppia/core/tests/test_utils.py", line 408, in get_json
    json_response, expect_errors=expect_errors)
  File "/Users/giritheja/Desktop/opensource/oppia/core/tests/test_utils.py", line 398, in _parse_json_response
    json_response.content_type, 'application/json')
AssertionError: 'text/html' != 'application/json'
`
I am not sure how to sort this out, so I used testapp.get